src/senaite/patient/tests/doctests/Patient.rst

```diff
@@ -46,3 +48,23 @@ A patient can have a primary and additional email addresses:
 
     >>> patient.getAdditionalEmails()
     [{'name': 'Work', 'email': 'wayne@example.com'}]
+
+Create patient with no MRN should fail:
+    >>> patient_api.is_patient_required()
+    True
+    >>> patient = api.create(patients, "Patient")
+    >>> patient
+    None
+    >>> plone_api.portal.set_registry_record("senaite.patient.require_patient", False)
+    >>> patient_api.is_patient_required()
+    False
+    >>> patient = api.create(patients, "Patient")
+    >>> patient
+    <Patient at /plone/patients/P000002>
```

Comment:
The validator is not called when the field is set. It is only called when the form is submitted, e.g. like `processForm` in AT but it is called `applyValue` here:
https://github.com/zopefoundation/z3c.form/blob/master/src/z3c/form/widget.py#L404

Therefore, this test will always pass unless you do the same validation dance inside the custom filed setter.